app/amity.py

```diff
@@ -1,33 +1,413 @@
-class Amity:
-    all_people = {}
-    amity_rooms = {}
-    office_allocations = {}
-    living_allocations = {}
+from collections import defaultdict
+import random
+import json
+from room import RoomClass, Office, Lspace
+from employee import PersonClass, Fellow, Staff
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+from database import Employee, Room, create_db, Base
+from sqlalchemy.sql import select
 
-    def create_room(self, room_name):
-        pass
+import random
 
-    def add_person(self):
-        pass
+class AmityDefaultDict(defaultdict):
+    def __setitem__(self, key, value):
+        key = key.upper()
+        dict.__setitem__(self, key, value)
 
-    def reallocate_person(self):
-        pass
+    def __getitem__(self, key):
+        key = key.upper()
+        return dict.__getitem__(self, key)
 
-    def load_people(self):
-        pass
 
-    def print_allocations(self):
-        pass
 
-    def print_unallocated(self):
-        unallocated = {}
-        pass
+class Amity(object):
+    def __init__(self):
+        self.all_people = []
+        self.all_rooms = []
+        self.office_allocations = AmityDefaultDict(list)
+        self.lspace_allocations = AmityDefaultDict(list)
+        self.amity_unallocated = []
+        self.fellows_list = []
+        self.staff_list = []
+        self.lspace_list = []
+        self.office_list = []
 
-    def print_all_rooms(self):
-        pass
+    def create_room(self, room_name, room_type):
+        """creates a room not previously in system"""
+        all_room_names = [room.room_name for room in self.all_rooms]
 
-    def save_state(self):
-        pass
+        if room_name.upper() in all_room_names:
+            print("sorry, room already exists!please choose another name")
 
-    def load_state(self):
-        pass
+        else:
+            if room_type.upper() == "LSPACE":
+                room = Lspace(room_name, room_type)
+                self.all_rooms.append(room)
+                print("room successfully created")
+            elif room_type.upper() == "OFFICE":
+                room = Office(room_name, room_type)
+                self.all_rooms.append(room)
+                print("room successfully created")
+            else:
+                print("sorry, that room_type does not exist")
+
+    def generate_room(self, r_type):
+        """returns a list of all rooms not full"""
+        available = []
+        for room in self.all_rooms:
+
+            if room.room_type.upper() == r_type.upper() and room.occupants < room.max_occupants:
+                if room.room_type.upper() =="LSPACE":
+                    max_occupants = 4
```

Comment:
Line 62 and 66 are not necessary i.e You already know max_occupants when getting into the logic and the variable is unused.